### ISSUE TITLE
Clicking on a PC while in Move mode in combat flees the battle

### ISSUE THREAD

**Issue by grannypron** [grannypron/uaf#31]
Tested in 5.21 with the default tutorial.

In combat, during a PC turn, press M to put the character in Move mode.  Use the mouse to move a square in some direction and then click directly on the PCs icon.  The character will flee the battle (Message "so and so has fled...")

**Comment by grannypron** [comment]
Stack trace:

```
UAFWin.exe!COMBATANT::HandleCurrState(int zeroMoveAttackOK) Line 3963	
UAFWin.exe!COMBAT_DATA::HandleCurrState(int zeroMoveAttackOK) Line 4894	 	UAFWin.exe!COMBAT_MOVE_MENU_DATA::OnKeypress(key_code key, char ascii) Line 21009
UAFWin.exe!TASKLIST::OnKeypress(KEY_DATA & kd) Line 836
UAFWin.exe!CProcessInput::ProcessInput(HWND__ * hwnd) Line 1150
```


This is because HandleCurrState is called and the Combatant's moveX and moveY are = -1 at this point.

**Comment by grannypron** [comment]
Does NOT happen when player presses Numpad 0 key, which is interesting because both the click and the Numpad 0 fall into the same case statement.   Need to determine the difference.

**Comment by grannypron** [comment]
Numpad 0 doesn't have a problem because when that is pressed, combatData.m_iMoveToX is set to -1 and combatData.m_iMoveToY is set to -1, but this is not the case when the mouse is clicked.  Therefore, the !ValidTerrainViewportCoords(combatData.m_iMoveToX,combatData.m_iMoveToY) check is passed instead of caught.

**Comment by grannypron** [comment]
What I would consider to be the simplest fix would be to throw a return statement in the if just like there is one in the if above it. 

```
    if (!ValidTerrainViewportCoords(combatData.m_iMoveToX,combatData.m_iMoveToY))
    {
      combatData.m_iMoveToX=-1;
      combatData.m_iMoveToY=-1;
      return; // PRS 20110502
      break;
    }
  if (   (combatData.getActiveCombatCharacter().x == combatData.m_iMoveToX)
        && (combatData.getActiveCombatCharacter().y == combatData.m_iMoveToY))
    {
      combatData.m_iMoveToX = -1;
      combatData.m_iMoveToY = -1;
      return; // return here - just like in the clause 8 lines above
      break;
    }

```

BUT

Before one moves a few paces, the behavior for clicking o a PC icon is to stop the move action and return to the regular combat menu.  If I put the change in above, clicking on the PC will not cancel the move operation, thereby changing the game's UI behavior, albeit slightly.

I investigated a bit to see the difference between before and after I moved a few paces and I can see that after a few paces, the State of the combatant is set to ICS_Moving, whereas in the former, it is set to ICS_None.  So, because I read the community to be averse to change in the UI behavior, I decided to instead just manually set the State() to ICS_None in the clause above.
